test_gracefull_death: Assertion error

continuation to #5742 found while testing #5735 with , while cmd:annex: 8.20201127 but not sure if related to having -J5 there or not

https://ci.appveyor.com/project/mih/datalad/builds/39550675/job/ns726sivdltwl9tu

```

======================================================================
FAIL: datalad.support.tests.test_parallel.test_gracefull_death
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/appveyor/dlvenv/lib/python3.8/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/appveyor/dlvenv/lib/python3.8/site-packages/datalad/tests/utils.py", line 266, in _wrap_skip_if
    return func(*args, **kwargs)
  File "/home/appveyor/dlvenv/lib/python3.8/site-packages/datalad/support/tests/test_parallel.py", line 197, in test_gracefull_death
    assert_greater_equal(len(results), 2)
AssertionError: 0 not greater than or equal to 2
```

most likely just fragility of that test

--- a/datalad/support/tests/test_parallel.py
+++ b/datalad/support/tests/test_parallel.py
@@ -179,7 +179,7 @@ def test_gracefull_death():
 
     def producer():
         for i in range(10):
-            sleep(0.0001)
+            sleep(0.0003)
             yield i
         raise ValueError()
     # by default we do not stop upon producer failing
@@ -196,10 +196,10 @@ def test_gracefull_death():
     assert_equal(results, list(range(len(results))))
     assert_greater_equal(len(results), 2)
     if info_log_level and not (on_windows or on_osx):
-        # windows does not behave according to the initial performance
-        # expectations gh-5296 (~9), and neither does a macosx cloud instance
-        # (~7)
-        assert_greater_equal(6, len(results))
+        # consumers should not be able to consume all produced items.
+        # production of 10 should take 3 unites, while consumers 10/2 (jobs)
+        # 5 units, so some should not have a chance.
+        assert_greater_equal(8, len(results))
 
     # Simulate situation close to what we have when outside code consumes
     # some yielded results and then "looses interest" (on_failure="error").

--- a/datalad/support/tests/test_parallel.py
+++ b/datalad/support/tests/test_parallel.py
@@ -195,11 +195,18 @@ def test_gracefull_death():
     # we will get some results, seems around 4 and they should be "sequential"
     assert_equal(results, list(range(len(results))))
     assert_greater_equal(len(results), 2)
-    if info_log_level and not (on_windows or on_osx):
-        # consumers should not be able to consume all produced items.
-        # production of 10 should take 3 unites, while consumers 10/2 (jobs)
-        # 5 units, so some should not have a chance.
-        assert_greater_equal(8, len(results))
+
+    # This test relies too much on threads scheduling to not hog up on handling
+    # consumers, but if it happens so - they might actually consume all results
+    # before producer decides to finally raise an exception.  As such it remains
+    # flaky and thus not ran, but could be useful to test locally while
+    # changing that logic.
+    #
+    # if info_log_level and not (on_windows or on_osx):
+    #     # consumers should not be able to consume all produced items.
+    #     # production of 10 should take 3 unites, while consumers 10/2 (jobs)
+    #     # 5 units, so some should not have a chance.
+    #     assert_greater_equal(8, len(results))
 
     # Simulate situation close to what we have when outside code consumes
     # some yielded results and then "looses interest" (on_failure="error").

